File: Examples/DotNET/CSharp/Pages/SignPage.cs
using System;
using Com.Aspose.PDF.Api;
using Com.Aspose.PDF.Model;
using Com.Aspose.Storage.Api;

namespace Pages
{
    class SignPage
    {
        public static void Run()
        {
            // ExStart:1
            PdfApi pdfApi = new PdfApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            string fileName = "sample-input.pdf";
            int? pageNumber = 1;
            string storage = null;
            string folder = null;

            Com.Aspose.PDF.Model.Signature body = new Com.Aspose.PDF.Model.Signature();
            body.Authority = "Authority";
            body.Contact = "naeem.akram@aspose.com";
            body.Date = "18-4-2016";
            body.FormFieldName = "Signature1";
            body.Location = "Gojra";
            body.Password = "aspose";
            Com.Aspose.PDF.Model.Rectangle rect = new Com.Aspose.PDF.Model.Rectangle();
            rect.X = 100;
            rect.Y = 100;
            rect.Height = 100;
            rect.Width = 200;
            body.Rectangle = rect;
            body.SignaturePath = "pkc7-sample.pfx";
            body.SignatureType = "PKCS7";
            body.Visible = true;

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));
                storageApi.PutCreate(body.SignaturePath, null, null, System.IO.File.ReadAllBytes(Common.GetDataDir() + body.SignaturePath)); 

                // Invoke Aspose.PDF Cloud SDK API to sign pdf page
                SaaSposeResponse apiResponse = pdfApi.PostSignPage(fileName, pageNumber, storage, folder, body);

                if (apiResponse != null && apiResponse.Status.Equals("Ok"))
                {                 
                    Console.WriteLine("Sign PDF Document Page, Done!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Node.js/Pages/SignPage.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var PdfApi = require('asposepdfcloud');
var StorageApi = require('asposestoragecloud');
var configProps = require('../config/config.json');
var data_path = '../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose.Pdf API SDK
var pdfApi = new PdfApi(config);

var name = "sample-input.pdf";
var signatureFileName = "pkc7-sample.pfx";
var pageNumber = 1;
var signatureBody = {
			'Authority' : 'Naeem Akram',
			'Location' : 'Gojra',
			'Contact' : 'naeem.akram@aspose.com',
			'Date' : '04/18/2017 2:00:00.000 AM',
			'FormFieldName' :  'Signature1',
			'Password' : 'aspose',
			'SignaturePath' : signatureFileName,
			'SignatureType' : 'PKCS7',
			'Visible' : true,
			'Rectangle' : {
				'X' : 100,
				'Y' : 100,
				'Height' : 100,
				'Width' : 200
			}
	}; 

try {
	// Upload source file to aspose cloud storage
	storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
	assert.equal(responseMessage.status, 'OK');
	
	storageApi.PutCreate(signatureFileName, null, null, data_path + signatureFileName , function(responseMessage) {	
	assert.equal(responseMessage.status, 'OK', '');
			
		pdfApi.PostSignPage(name, pageNumber, null, null, signatureBody, function(responseMessage) {
				assert.equal(responseMessage.status, 'OK', '');
				console.log('body:', responseMessage.body);					
		});
	});
});

}
catch (e) 
{
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1